src/common/util.cc

```diff
@@ -109,6 +110,15 @@ static bool os_release_parse(map<string, string> *m, CephContext *cct)
   file_values_parse(kvm, fp, m, cct);
 
   fclose(fp);
+#elif defined(__FREEBSD__)
+  struct utsname u;
+  r = uname(&u);
```

Comment:
where is `r` defined? 

and how did you test this patch?

Reply:
@tchaikov 
I normally load the patch into my Jenkins testing, which catches these errors.
In this case I used the wrong define, which I corrected without updating the diff in my testruns.
So that code excluded both the Linux branch and the FreeBSD branch.

Reply:
but that does not explain the missing `r`. i'd suggest you recheck the macro to ensure your change actually takes effect on FreeBSD.